### PR TITLE
Multiloader using Architectury

### PR DESCRIPTION
**Pull request by SLimeyMC**
# Gradle Task

- [x] Folder structure
- [x] Root gradle
- [x] Common gradle
- [x] Fabric gradle
- [x] Forge gradle
- [x] Version catalog (?) (ditch version catalog for now)

# Source Task
 
- [ ] Networking
- [ ] Entity
- [ ] GUI
- [ ] Config
- [ ] EventHandler
- [ ] Renderer
- [ ] Fabric Platform
- [ ] Forge Platform
- [ ] Fabric Mod Compat
- [ ] Forge Mod Compat

### CODE REVIEW

**Comment by getItemFromBlock** [comment]
Thanks for the PR, there is still quite a lot left to do but that should help towards a fabric version
I will take a closer look this weekend

**Comment by getItemFromBlock** [comment]
After a quick look there are some things that I don't understand completely, like why do we need to have the assets present in both the forge and fabric part, and not in the common part ?

**Comment by SLimeyMC** [comment]
Oh yeah, I dont know what I was thinking when separating it, I will work on putting it in common 😅 

Anyway, I will be abandoning version catalog, it caused a problem with Gradle. 
Also, do you want the Gradle to be written in Kotlin or not?

**Comment by SLimeyMC** [comment]
Anyway i will be abandoning version catalog, it caused problem with gradle. 
Also, do you want the gradle to be written in kotlin or not?

**Comment by getItemFromBlock** [comment]
I am not really familiar with Kotlin (hence why my mod is made in Java) so preferably not, but if it makes things easier to you then go for it
Actually I don't even know what would be the alternative, is it like the original build.gradle file ?

**Comment by SLimeyMC** [comment]
Nope, wildly different. Forge uses Forgegradle which has a lot of kinks IMO.

**Comment by getItemFromBlock** [comment]
Alright, then let's go with Kotlin

**Comment by getItemFromBlock** [comment]
Hi, any news on this ?
Tell me when you have finished all the gradle/initial setup part so that I can continue to work on this

**Comment by SLimeyMC** [comment]
Alright, I already smashed alot of bugs and it can build now. You may code peacefully

For context, I have separated the task into gradle and source tasks. Hopefully, the added task reflects what needs to be done

**Comment by getItemFromBlock** [comment]
Thanks a lot, this is some really good avancement for fabric compat
I will take a closer look when things get less busy for me

**Comment by Frankky001** [comment]
is the fabric port still being worked on? would be really nice if it releases before create aero. no pressure ofc 😊

**Comment by getItemFromBlock** [comment]
Yes, I plan to continue working on this soon
Right now I am taking some well deserved vacations, but don't worry I have not forgotten about this
Note that the Architectury port is not the only thing planned

**Comment by AnCarsenat** [comment]
Wow so nice ! Hope you get it down ! Great Work !
Also, I had an issue open with "fabric pls" im so sorry, I did not know
there was already a fork open ....

**Comment by getItemFromBlock** [comment]
Update on this, for the people that want fabric:
I pushed the some old changes that I should have pushed a while ago on a new branch "dev-1.18-architectury"
The mod does not build correctly. For some reasons gradle refuses to include the mod's code in the jar, and my lack of knowledge in gradle build script does not help
If someone want to take a look at the problem I would be very grateful

**Comment by SLimeyMC** [comment]
Turn it into a PR draft then, I'll look into it later. May as well link it here and close this one.

**Comment by SLimeyMC** [comment]
Sorry, let me rephrase. Make a new PR draft using the new branch you created, then post a comment linking to the new PR and close this PR.

**Comment by getItemFromBlock** [comment]
Oh mb

**Comment by getItemFromBlock** [comment]
Moved to #42
